runtime/executor/src/main/java/org/apache/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -123,9 +133,28 @@ public TaskExecutor(final Task task,
     this.dataFetchers = pair.left();
     this.sortedHarnesses = pair.right();
 
+    this.streamMetricMap = new HashMap<>();
+    for (final DataFetcher dataFetcher : dataFetchers) {
+      String sourceVertexId = dataFetcher.getDataSource().getId();
+      this.streamMetricMap.put(sourceVertexId, new StreamMetric(sourceVertexId));
+    }
+
+    if (streamMetricPeriod > 0) {
+      this.periodicMetricService = Executors.newScheduledThreadPool(1);
+      this.periodicMetricService.scheduleAtFixedRate(this::saveMetric, 0, streamMetricPeriod, TimeUnit.MILLISECONDS);
+    }
     this.timeSinceLastExecution = System.currentTimeMillis();
   }
 
+  // Send stream metric to the runtime master
+  private void saveMetric() {
```

Comment:
Shouldn't the counter be reset upon every reset, in order to calculate the # of elements per streamMetricPeriod?

Reply:
Also, I think the name should be `saveStreamMetric()`